cirq/protocols/decompose.py

```diff
@@ -234,8 +239,10 @@ def decomposer(op):
         if keep is not None and on_stuck_raise is not None:
             if isinstance(on_stuck_raise, Exception):
                 raise on_stuck_raise
-            else:
-                raise on_stuck_raise(item)
+            elif callable(on_stuck_raise):
+                error = on_stuck_raise(item)
+                if isinstance(error, Exception):
```

Comment:
Change this to `if error is not None`. I would rather the code attempt to raise bad objects rather than silently keep going.

Reply:
Done